src/util/prefix_filter.cpp

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************\
+
+Module: Prefix Filtering
+
+Author: Peter Schrammel
+
+\*******************************************************************/
+
+/// \file
+/// Prefix Filtering
+
+#include "prefix_filter.h"
+
+#include <algorithm>
+
+#include "prefix.h"
+
+prefix_filtert::prefix_filtert(
+  std::vector<std::string> included_prefixes,
+  std::vector<std::string> excluded_prefixes)
+  : included_prefixes(std::move(included_prefixes)),
+    excluded_prefixes(std::move(excluded_prefixes))
+{
+}
+
+bool prefix_filtert::operator()(const std::string &value)
+{
+  if(!included_prefixes.empty())
```

Comment:
Make `is_included` and `is_excluded` their own functions, then you can write like
```
return included_prefixes.empty() || std::any_of(...)
```
and the main function can be clearer: `is_included(value) && !is_excluded(value)`